### PR TITLE
react2-week3

### PR DESCRIPTION
**Pull request by Hadis-jamali**

### CODE REVIEW

File: frontend/src/components/ContactUs/ContactUs.jsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import React from "react";
+import { Link } from "react-router-dom";
+import "./ContactUs.css";
+// import foodGif from "../../assets/images/food.gif";
```
Comment (BrantalikP) [inline frontend/src/components/ContactUs/ContactUs.jsx]: remove the unsused imports

File: backend/src/backend/api/meals.js [hunk 2]
```diff
@@ -93,13 +100,4 @@ router.get("/:id", async (req, res) => {
   }
 });
 
-// router.get("/:id/reviews", async (req, res) => {
```
Comment (BrantalikP) [inline backend/src/backend/api/meals.js:96]: if the code is not used, please remove it

File: frontend/src/components/MealsList.jsx [hunk 3]
```diff
@@ -43,11 +43,12 @@ function MealsList() {
   return (
     <>
       <Search search={search} setSearch={setSearch} searchHandler={searchHandler} />
+      
       <div className="meal-list">
         {loading ? (
           <p>Loading...</p>
         ) : meals.data.length > 0 ? (
-          meals.data.map((meal) => <MealItem key={meal.meal_id} meal={meal} />)
+          meals.data.map((meal,index) => <MealItem key={index} meal={meal} />)
```
Comment (BrantalikP) [inline frontend/src/components/MealsList.jsx]: its not considered a good practice to use index for `key`, please change to id ( you are using it in the `MealItem` already)

File: frontend/src/components/Modal/Modal.css [hunk 4]
```diff
@@ -22,18 +22,19 @@
 }
 
 .close {
-  top: 5px;
-  right: 5px;
+  /* top: 4px; */
+  /* right: 5px; */
```
Comment (BrantalikP) [inline frontend/src/components/Modal/Modal.css]: unused code, please remove it

File: frontend/src/components/MenuPage/MenuPage.jsx [hunk 5]
```diff
@@ -8,7 +8,7 @@ function MenuPage() {
       </div>
       <div className="menu-container">
         <h1>Menu</h1>
-        <p>Available meals in our place</p>
+        <p className="Available">Available meals in our place</p>
```
Comment (BrantalikP) [inline frontend/src/components/MenuPage/MenuPage.jsx]: please keep the naming consistent :) Lets use lowercase letter at the beginning please

File: frontend/src/components/ContactUs/ContactUs.jsx [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+function ContactUs() {
+  return (
+    <div className="about">
+      <div className="main-about">
+        <div className="animation"></div>
```
Comment (BrantalikP) [inline frontend/src/components/ContactUs/ContactUs.jsx]: you can use self-closing tags by `<div /> `

File: frontend/src/components/Modal/Modal.jsx [hunk 7]
```diff
@@ -4,14 +4,14 @@ import { Link } from "react-router-dom";
 function Modal() {
   return (
     <div>
```
Comment (BrantalikP) [inline frontend/src/components/Modal/Modal.jsx]: is div needed? I think you can remove it

File: frontend/src/components/Rating/Review.jsx [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+      });
+  };
+
+  return (
+    <div>
+      <div className="meal-star">
+        {[...Array(5)].map((_, index) => {
+          const starValue = index + 1;
+          return (
+            <label key={index}>
```
Comment (BrantalikP) [inline frontend/src/components/Rating/Review.jsx]: again please replace the index with something less generic

File: frontend/src/components/Rating/Review.jsx [hunk 9]
```diff
@@ -0,0 +1,91 @@
+import React from "react";
+import { FaStar } from "react-icons/fa";
+import { useState } from "react";
+import "./Review.css";
+function Review({ mealId }) {
+  const [rating, setRating] = useState(null);
+  const [hover, setHover] = useState(null);
+  let currentRating = 1;
```
Comment (BrantalikP) [inline frontend/src/components/Rating/Review.jsx]: as the rating never will be changed it should be const and also as the static value we can move it outside of the React render :) 
```
const currentRating = 1;
function Review({ mealId }) {
  const [rating, setRating] = useState(null);
```

File: frontend/src/components/Rating/Review.jsx [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+    title: "",
+  });
+
+  const clickHandler = (event) => {
+    event.preventDefault();
+
+    const starValue = Number(event.target.value);
+    setRating(starValue);
+
+    const meal_id = mealId;
```
Comment (BrantalikP) [inline frontend/src/components/Rating/Review.jsx]: Is there a special reason why to reassign the `mealId` to a new variable? I think you can use `mealId` as is, no?

File: frontend/src/components/Rating/Review.jsx [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+    // const title = data.data[0].title;
+    const create_date = new Date().toISOString().split("T")[0];
+    const reviewData = JSON.stringify({
+      ...newReview,
+      stars: currentRating,
+      meal_id: Number(meal_id),
+      create_date: create_date,
+    //   title: title,
```
Comment (BrantalikP) [inline frontend/src/components/Rating/Review.jsx]: if it's not use, lets remove it, or you can use a special syntax to comment code that will be needed later like 
// FIXME: "some comment" or 
// TODO: "some comment"

Those two are really common among developers :)

File: frontend/src/components/Rating/Review.jsx [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+          // title: "",
+        });
+      })
+      .catch((error) => {
+        console.error("Error submitting reservation:", error);
+      });
+  };
+
+  return (
+    <div>
```
Comment (BrantalikP) [inline frontend/src/components/Rating/Review.jsx]: I think we can remove the extra div :)

File: frontend/src/components/Rating/Review.jsx [hunk 13]
```diff
@@ -0,0 +1,91 @@
...
+        }
+        return response.json();
+      })
+      .then((data) => {
+        console.log("Reservation successful:", data);
+        setNewReview({
+          meal_id: "",
+          stars: "",
+          create_date: "",
+          // title: "",
```
Comment (BrantalikP) [inline frontend/src/components/Rating/Review.jsx]: same as above:}

**Review by BrantalikP — CHANGES_REQUESTED** [review]
Hey @Hadis-jamali well done 👏 I've left a few comments but most of them are just about removing unused code. I know this might feel a bit silly while you are working alone on the project but you find a benefit of keeping the codebase clean and easy to comprehend once you join teams of bigger scale :)

File: frontend/src/components/MealsList.jsx (hunk 3 above)
Comment (Hadis-jamali) [reply frontend/src/components/MealsList.jsx]: done!

File: frontend/src/components/Rating/Review.jsx (hunk 10 above)
Comment (Hadis-jamali) [reply frontend/src/components/Rating/Review.jsx]: I don't know why!! I changed.

**Pull request merged**